src/lib/clients/phony.ts

```diff
@@ -1,21 +1,69 @@
-import { BaseOptions } from "../../types/generic";
-import { Client } from "../../types/generic";
+import { BaseOptions, HandlerBody } from '../../types/generic';
+import { SayArguments } from "@slack/bolt";
+import { KeeperSlackMiddleware } from "../../types/clients/keeper"
+import { CommandClient, Command } from '../../types/generic';
 import { Logger } from '../../types/logger'
+import { EventEmitter } from 'events';
+import * as net from 'net';
 
-export default class PhonyClient implements Client {
+export default class PhonyClient extends EventEmitter implements CommandClient<PhonyClient> {
     private logger: Logger;
+    private server: net.Server;
+    private commands: Command<KeeperSlackMiddleware, this>[];
 
-    constructor (options: BaseOptions) {
+    constructor(options: BaseOptions) {
+        super();
+        this.commands = [];
         this.logger = options.logger;
+        this.server = net.createServer(c => {
+            c.on('connect', () => this.logger.log('tcp connection opened'));
+            c.on('end', () => this.logger.log('tcp connection closed'));
+
+            c.on('data', async (data) => {
+                const msg = data.toString().trim();
+
+                await Promise.all(this.commands.filter(({ matches }) => {
+                    switch (typeof matches) {
+                        case "object": return matches.test(msg);
+                        case "string": return matches == msg;
+                    }
+                }).map(({ handler }) => {
+                    return handler(this.handlerArgs(c, msg))
+                }));
+            });
+        });
+    }
+
+    handlerArgs(c: net.Socket, text: string): HandlerBody<KeeperSlackMiddleware, this> {
+        return {
+            client: this,
+            logger: this.logger.log,
```

Comment:
Do we purposely pass `this.logger.log` or should we just be passing `this.logger`?

Reply:
The `HandlerBody.logger` type is `Function` so this was necessary, I do agree it's probably better to just attach the whole `logger` though, I'll get that changed